fiftyone/core/stages.py

```
@@ -389,6 +448,133 @@ class ViewStageError(Exception):
+# The schema a field constraint draws from
+_ANY_LEVEL = "any"
+_SAMPLE_LEVEL = "sample"
+_FRAME_LEVEL = "frame"
+
+# Whether a field constraint accepts a name that does not exist yet
+_EXISTING = "existing"
+_EXISTING_ROOT = "existing_root"
+_ANY_NAME = "any"
+
+
+def _type_paths(types):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `_type_paths`

(ANN202)